File: JavaDemo/SpringBootDemo/src/main/java/com/example/demo/service/UserService.java
package com.example.demo.service;

import com.example.demo.entity.User;
import com.example.demo.repository.UserRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import javax.transaction.Transactional;

@Transactional
@Service
@RequiredArgsConstructor(onConstructor = @__(@Autowired))
public class UserService {
    private final UserRepository repository;

    public boolean exists(User user){
        return repository.existsByName(user.getName());
    }

    public User findByNameAndPassword(User user){
        return repository.findByNameAndPassword(user.getName(),user.getPassword());
    }

    public boolean insert(User user){
        repository.save(user);
        return true;
    }

    public boolean update(User user){
        if(repository.findById(user.getId()).isEmpty()){
            return false;
        }
        repository.save(user);
        return true;
    }

    public boolean deleteById(int id){
        if(!repository.existsById(id)){
            return false;
        }
        repository.deleteById(id);
        return true;
    }
}

File: KotlinDemo/SpringBootDemo/src/main/kotlin/com/example/demo/service/UserService.kt
package com.example.demo.service

import com.example.demo.entity.User
import com.example.demo.repository.UserRepository
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service
import javax.transaction.Transactional

@Transactional
@Service
class UserService {
    @Autowired
    lateinit var repository:UserRepository

    fun exists(user: User) = repository.existsByName(user.name)

    fun findByNameAndPassword(user:User) = repository.findByNameAndPassword(user.name,user.password)

    fun insert(user:User):Boolean{
        repository.save(user)
        return true
    }

    fun update(user:User):Boolean{
        return if(repository.findById(user.id).isEmpty){
            false
        }else{
            repository.save(user)
            true
        }
    }

    fun deleteById(id:Int):Boolean{
        return if(!repository.existsById(id)) {
            false
        }else{
            repository.deleteById(id)
            true
        }
    }
}